tests/Compilation/positive/issue1706.juvix

```diff
@@ -0,0 +1,10 @@
+module issue1706;
+  open import Stdlib.Prelude public;
+
+  f : Nat → Nat → Nat;
```

Comment:
Hypothesis: This will break if the `f (suc n)` appears first because the internal to core translation uses the head clause to determine the number of arguments (similar issue for lambda clauses). 

So add a test where the first clause has fewer arguments that a subsequent clause.